adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala

```diff
@@ -339,6 +339,373 @@ private[adam] class VariantContextConverter extends Serializable with Logging {
     ???
   }
 
+  // variant conversion functions
+
+  private[converters] def formatNames(
+    vc: HtsjdkVariantContext,
+    vb: Variant.Builder): Variant.Builder = {
+
+    splitIds(vc).foreach(vb.setNames(_))
+    vb
+  }
+
+  private[converters] def formatFilters(
+    vc: HtsjdkVariantContext,
+    vb: Variant.Builder): Variant.Builder = {
+
+    vb.setFiltersApplied(vc.filtersWereApplied)
+    if (vc.filtersWereApplied) {
+      vb.setFiltersPassed(!vc.isFiltered)
+    }
+    if (vc.isFiltered) {
+      vb.setFiltersFailed(new java.util.ArrayList(vc.getFilters));
+    }
+    vb
+  }
+
+  private[converters] def formatSomatic(
+    vc: HtsjdkVariantContext,
+    vb: Variant.Builder): Variant.Builder = {
+
+    Option(vc.getAttribute("SOMATIC").asInstanceOf[java.lang.Boolean]).foreach(vb.setSomatic(_))
```

Comment:
Do you know how the flag fields are stored when parsing from VCF? I had a nasty surprise WRT this and the other attribute fields. I haven't looked ahead, but if you haven't added this yet, I would add a unit test with a VCF with a somatic call.